Cognite.Simulator.Tests/UtilsTests/TestUtilities.cs

```diff
@@ -101,24 +102,67 @@ public static Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessa
                     throw new ArgumentNullException(nameof(uri));
                 }
 
-                foreach (var mapping in endpointMappings)
+                foreach (var mockTemplate in endpointMockTemplates)
                 {
-                    if (mapping.Key(uri))
+                    if (mockTemplate.Matches(uri))
                     {
-                        var (responseFunc, callCount, maxCalls) = mapping.Value;
-                        if (maxCalls.HasValue && callCount >= maxCalls)
-                        {
-                            return GoneResponse;
-                        }
-                        endpointMappings[mapping.Key] = (responseFunc, callCount + 1, maxCalls);
-                        return responseFunc();
+                        return mockTemplate.GetResponse();
                     }
                 }
-
-                return new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent("{\"items\":[]}") };
+                return CreateResponse(HttpStatusCode.NotImplemented, "Not implemented");
             };
         }
 
+        /// <summary>
+        /// A container for easier mocking of endpoints.
+        /// </summary>
+        public class SimpleRequestMocker
+        {
+            private readonly Func<string, bool> _uriMatcher;
+            private readonly Func<HttpResponseMessage> _responseFunc;
+            private int _callCount = 0;
```

Comment:
Would exposing this be useful for our tests?

Reply:
I exposed something else now => `AssertCallCount` should be much more useful